src/flow/src/batching_mode/frontend_client.rs

```diff
@@ -216,7 +220,13 @@ impl FrontendClient {
             {
                 let addr = &node_info.peer.addr;
                 let client = Client::with_manager_and_urls(chnl_mgr.clone(), vec![addr.clone()]);
-                let database = Database::new(catalog, schema, client);
+                let database = {
+                    let mut db = Database::new(catalog, schema, client);
+                    if let Some(auth) = auth {
+                        db.set_auth(auth.auth().clone());
```

Comment:
Is this `clone` invoked on every flow request in the hot path?

Reply:
> Is this `clone` invoked on every flow request in the hot path?

not really a hot path since already wait on many thing else, so clone is acceptable